doc/sphinx/proof-engine/tactics.rst

```diff
@@ -2285,6 +2286,18 @@ and an explanation of the underlying technique.
       to the number of new equalities. The original equality is erased if it
       corresponds to a hypothesis.
 
+   .. tacv:: injection @term {? with @bindings_list} as @injection_intropattern
+             injection @num as @injection_intropattern
+             injection as @injection_intropattern
+             einjection @term {? with @bindings_list} as @injection_intropattern
+             einjection @num as @injection_intropattern
+             einjection as @injection_intropattern
+
+      These are equivalent to the previous variants but using instead the
+      syntax :token:`injection_intropattern` which :tacn:`intros`
+      uses. In particular :n:`as [= {+ @simple_intropattern}]` behaves
+      the same as :n:`as {+ @simple_intropattern}`.
```

Comment:
Is the `[= ]` pattern supposed to contain a list of `simple_intropattern` or a list of `intropattern`? There seems to be a contradiction between the line above and the grammar definition.

Reply:
`[= ]` contains a list of `intropattern` but, in the other syntax, `injection term as simple_intropattern_list` these are restricted to `simple_intropattern` (not `intropattern`). So the story is a bit complicated.

Maybe can I add in the documentation that, in addition, `[= ]`  accepts `*` and `**` which span until the end of the block of injection equations under consideration?

Otherwise, an alternative would be to accept `*` and `**` in the `injection term as simple_intropattern_list` syntax. I'm a bit reluctant because I'm a bit afraid that it is not as obvious that the scope of `*` and `**` shall be the block of injection equations under consideration.

What would you suggest?

Reply:
What is the preference to give between `` `intropattern` ... `intropattern` `` and `` `{+ @intropattern}` ``? The doc seems to use both.

Reply:
In `productionlist` we don't have a choice and are forced to use the first one, but generally we prefer the second when we can.